zookeeper-server/src/test/java/org/apache/zookeeper/server/CreateContainerTest.java

```diff
@@ -237,6 +237,22 @@ public Void call() throws Exception {
         assertEquals(queue.poll(5, TimeUnit.SECONDS), "/four");
     }
 
+    @Test(timeout = 30000)
+    public void testContainerWithNoChildGracePeriod() throws KeeperException, InterruptedException {
+        zk.create("/foo", new byte[0], ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.CONTAINER);
+
+        ContainerManager containerManager = new ContainerManager(serverFactory.getZooKeeperServer().getZKDatabase(), serverFactory.getZooKeeperServer().firstProcessor, 1, 100);
+        containerManager.checkContainers();
+        Thread.sleep(1000);
+
+        assertNotNull("Container should still be there", zk.exists("/foo", false));
+
+        containerManager.checkContainers();
```

Comment:
Can we avoid this sleep?
This test may turn into a flaky one

Reply:
I used the same approach as other tests in that class, where I understand that these `sleep` after each call to `checkContainers` are precisely there to avoid flakiness by giving time to the ZK server to asynchronously delete the nodes that have been selected for removal.

Did I miss something?